libs/sdk-ui-pivot/src/impl/structure/colDefFactory.ts

```diff
@@ -282,7 +315,10 @@ export function createColDefsFromTableDescriptor(
 
     createAndAddSliceColDefs(table.sliceCols, table.sliceMeasureCols, state);
     createAndAddDataColDefs(table, state, intl);
-    // handle metrics in rows and no column attribute case
+    if (config?.columnHeadersPosition === "left") {
```

Comment:
on other places, the check also includes the isTransposed() part. Consider adding it here and you may even remove the isTransposed from sending to the method and remove the if statement here https://github.com/gooddata/gooddata-ui-sdk/pull/3865/files#diff-ca12b4453929dd05baa87f0e6b40f136da5fdbacf1d25985af91612288bb3641R133

Reply:
explained by comment